File: app-flutter/lib/router/uiRouter.dart
import 'package:flutter_web/material.dart';
// ======================================================================================================== 路由
import 'package:flutter_web_study.app/pages/flutter/ui/Card.dart';
import 'package:flutter_web_study.app/pages/flutter/ui/Chip.dart';
import 'package:flutter_web_study.app/pages/flutter/ui/Tooltip.dart';
import 'package:flutter_web_study.app/pages/flutter/ui/DataTable.dart';
import 'package:flutter_web_study.app/pages/flutter/ui/LinearProgressIndicator.dart';
import 'package:flutter_web_study.app/pages/flutter/ui/CircularProgressIndicator.dart';
import 'package:flutter_web_study.app/pages/flutter/ui/RefreshIndicator.dart';
import 'package:flutter_web_study.app/pages/flutter/ui/Divider.dart';
import 'package:flutter_web_study.app/pages/flutter/ui/Stepper.dart';
import 'package:flutter_web_study.app/pages/flutter/ui/UserAccountsDrawerHeader.dart';
import 'package:flutter_web_study.app/pages/flutter/ui/FlutterLogo.dart';
import 'package:flutter_web_study.app/pages/flutter/ui/Placeholder.dart';
// 标题
import 'package:flutter_web_study.app/pages/flutter/ui/title/Title.dart';
import 'package:flutter_web_study.app/pages/flutter/ui/title/ListTile.dart';
import 'package:flutter_web_study.app/pages/flutter/ui/title/ExpansionTile.dart';
import 'package:flutter_web_study.app/pages/flutter/ui/title/AboutListTile.dart';
// tab
import 'package:flutter_web_study.app/pages/flutter/ui/tab/BottomNavigationBar.dart';
import 'package:flutter_web_study.app/pages/flutter/ui/tab/TabBar.dart';
import 'package:flutter_web_study.app/pages/flutter/ui/tab/TabBarView.dart';
// ======================================================================================================== x

// ======================================================================================================== 路由类
class UiRouter{
	final Map<String,WidgetBuilder> router = <String,WidgetBuilder>{
		'/CardPage': (BuildContext context) => CardPage(title:'Card( )'),
		'/ChipPage': (BuildContext context) => ChipPage(title:'Chip( )'),
		'/TooltipPage': (BuildContext context) => TooltipPage(title:'Tooltip( )'),
		'/DataTablePage': (BuildContext context) => DataTablePage(title:'DataTablePage( )'),
		'/LinearProgressIndicatorPage': (BuildContext context) => LinearProgressIndicatorPage(title:'LinearProgressIndicator( )'),
		'/CircularProgressIndicatorPage': (BuildContext context) => CircularProgressIndicatorPage(title:'CircularProgressIndicator( )'),
		'/RefreshIndicatorPage': (BuildContext context) => RefreshIndicatorPage(title:'RefreshIndicator( )'),
		'/DividerPage': (BuildContext context) => DividerPage(title:'Divider( )'),
		'/StepperPage': (BuildContext context) => StepperPage(title:'Stepper( )'),
		'/UserAccountsDrawerHeaderPage': (BuildContext context) => UserAccountsDrawerHeaderPage(title:'UserAccountsDrawerHeader( )'),
		'/FlutterLogoPage': (BuildContext context) => FlutterLogoPage(title:'FlutterLogo( )'),
		'/PlaceholderPage': (BuildContext context) => PlaceholderPage(title:'Placeholder( )'),
		// 标题
		'/TitlePage': (BuildContext context) => TitlePage(title:'Title( )'),
		'/ListTilePage': (BuildContext context) => ListTilePage(title:'ListTile( )'),
		'/ExpansionTilePage': (BuildContext context) => ExpansionTilePage(title:'ExpansionTile( )'),
		'/AboutListTilePage': (BuildContext context) => AboutListTilePage(title:'AboutListTile( )'),
		// tab
		'/BottomNavigationBarPage': (BuildContext context) => BottomNavigationBarPage(title:'BottomNavigationBar( )'),
		'/TabBarPage': (BuildContext context) => TabBarPage(title:'TabBar( )'),
		'/TabBarViewPage': (BuildContext context) => TabBarViewPage(title:'TabBarView( )'),
	};
}

File: app-react/src/pages/flutter/routers/uiRouter.js
/* ====================================== 模块子路由配置  ====================================== */
import React from 'react'
import { Switch, Route } from 'react-router-dom'
//import { TransitionGroup, CSSTransition } from 'react-transition-group'
// ===================================================================== 异步加载
import { Import } from '@common/bundle'
// ===================================================================== 二级路由
export default class 
 extends React.Component{
	render(){
		const { match } = this.props
		return (
				<Switch>
					<Route path={`${match.url}/Card`} 	component={ Import('flutter/ui/Card') } exact />
					<Route path={`${match.url}/Chip`} 	component={ Import('flutter/ui/Chip') } exact />
					<Route path={`${match.url}/Tooltip`} 	component={ Import('flutter/ui/Tooltip') } exact />
					<Route path={`${match.url}/DataTable`} 	component={ Import('flutter/ui/DataTable') } exact />
					<Route path={`${match.url}/RefreshIndicator`} 	component={ Import('flutter/ui/RefreshIndicator') } exact />
					<Route path={`${match.url}/Divider`} 	component={ Import('flutter/ui/Divider') } exact />
					<Route path={`${match.url}/Stepper`} 	component={ Import('flutter/ui/Stepper') } exact />
					<Route path={`${match.url}/UserAccountsDrawerHeader`} 	component={ Import('flutter/ui/UserAccountsDrawerHeader') } exact />
					<Route path={`${match.url}/FlutterLogo`} 	component={ Import('flutter/ui/FlutterLogo') } exact />
					<Route path={`${match.url}/Placeholder`} 	component={ Import('flutter/ui/Placeholder') } exact />
					{/* 标题  */}
					<Route path={`${match.url}/Title`} 	component={ Import('flutter/ui/title/Title') } exact />
					<Route path={`${match.url}/ListTile`} 	component={ Import('flutter/ui/title/ListTile') } exact />
					<Route path={`${match.url}/ExpansionTile`} 	component={ Import('flutter/ui/title/ExpansionTile') } exact />
					<Route path={`${match.url}/AboutListTile`} 	component={ Import('flutter/ui/title/AboutListTile') } exact />
					{/* tab */}
					<Route path={`${match.url}/BottomNavigationBar`} 	component={ Import('flutter/ui/tab/BottomNavigationBar') } exact />
					<Route path={`${match.url}/TabBar`} 	component={ Import('flutter/ui/tab/TabBar') } exact />
					<Route path={`${match.url}/Tab`} 	component={ Import('flutter/ui/tab/Tab') } exact />
					<Route path={`${match.url}/TabBarView`} 	component={ Import('flutter/ui/tab/TabBarView') } exact />
					{/* 进度条 */}
					<Route path={`${match.url}/LinearProgressIndicator`} 	component={ Import('flutter/ui/progress/LinearProgressIndicator') } exact />
					<Route path={`${match.url}/CircularProgressIndicator`} 	component={ Import('flutter/ui/progress/CircularProgressIndicator') } exact />
				</Switch>
		)
	}
}